initializr-generator/src/main/groovy/io/spring/initializr/generator/ProjectRequest.groovy

```diff
@@ -78,14 +78,30 @@ class ProjectRequest extends BasicProjectRequest {
 	 */
 	void resolve(InitializrMetadata metadata) {
 		List<String> depIds = style ? style : dependencies
+		List<String> additionalDepIds = new ArrayList<>()
 		String actualBootVersion = bootVersion ?: metadata.bootVersions.default.id
 		Version requestedVersion = Version.parse(actualBootVersion)
 		resolvedDependencies = depIds.collect {
 			def dependency = metadata.dependencies.get(it)
 			if (dependency == null) {
 				throw new InvalidProjectRequestException("Unknown dependency '$it' check project metadata")
 			}
-			dependency.resolve(requestedVersion)
+			dependency = dependency.resolve(requestedVersion)
```

Comment:
That's no really what I have in mind for this feature. What you've implemented looks dangerously like "profile" (i.e. you select one box and several boxes are added). What I had in mind was a way to select a box and have several dependencies added to the project (without being displayed upfront). So you select "foo", you only show "foo" but "foo" and "bar" are added to the build. "bar" can't be selected.

Implementing this in the form of a `Dependency` list with some smart default inheritance should do the trick. Now I understand that's not what you're proposing here and I am not keen to merge it. I guess that's not helping since your fork is using the feature as is at the moment. 

What could we do to conciliate those use cases?